### DISCUSSION TITLE
Moving from Axios to a native fetch API

### DISCUSSION THREAD

**Discussion by capsloq** [refinedev/refine#6375] [General]
What is Refine's view towards retiring Axios and moving torwards a native fetch API or a thin wrapper like `KY`?
Axios is considered deprecated by (many/some) however it's at the very core of Refine.
Mainly using the arguments of [adios-axios](https://www.adios-axios.com/).

TL;DR Quote:

> Axios was once a convenient library that utilized [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) in web browsers and the [http](https://nodejs.org/api/http.html) module in Node environments. However, in today's modern development landscape, this approach has been surpassed by the native Fetch API, which offers consistent functionality across browsers, Node, and even other environments.

> With the adoption of a unified API like Fetch, you can easily eliminate Axios as an additional dependency in your modern codebase, resulting in reduced bundle sizes. Moreover, the Fetch API has undergone full standardization, unlike Axios, which comes with its own set of nuances that often catch developers off-guard. This includes differences in how fetch and Axios handle promise rejection in various scenarios.

> Furthermore, certain frameworks, like Next.js, have begun exposing additional features within the Fetch API that may not be readily accessible when using 

Any plans / views regarding this for the future?

**Comment by alicanerdurmaz** [comment]
We don’t use Axios in `@refinedev/core` or on the tutorial, but it is used in some documentation examples and data providers. Honestly, Refine doesn't have an opinion on which HTTP client to use, you can use whichever you prefer, but since we use Axios in some data providers, it might seem that way.

We don't plan to switch from Axios to native fetch API right now but feel free to open a PR. For example, you could add `ky` support under `@refinedev/simple-rest/ky`.

**Reply by capsloq** [reply]
Thank you for the clarification. I was indeed under the impression that Axios is used in the core after looking at the data providers.
I will see what I can do and get a PR started.

**Reply by alicanerdurmaz** [reply]
Thanks for improving Refine 🚀 

Could you please open an issue first? This way, we can track the task more easily.

**Reply by capsloq** [reply]
done, feel free to close this dicussion
